gdal/ogr/ogrsf_frmts/gpkg/ogrgeopackagetablelayer.cpp

```diff
@@ -933,6 +933,39 @@ OGRErr OGRGeoPackageTableLayer::ReadTableDefinition()
                     OGRSpatialReference *poSRS = m_poDS->GetSpatialRef(m_iSrs);
                     if ( poSRS )
                     {
+                        // Retrieve coordinate epoch
+                        if( m_poDS->HasMetadataTables() )
+                        {
+                            pszSQL = sqlite3_mprintf(
+                                "SELECT md.metadata FROM gpkg_metadata md "
+                                "JOIN gpkg_metadata_reference mdr ON (md.id = mdr.md_file_id ) "
+                                "WHERE md.metadata LIKE 'coordinate_epoch=%%' AND "
+                                "md.md_standard_uri = 'http://gdal.org' AND "
+                                "md.mime_type = 'text/plain' AND "
+                                "lower(mdr.table_name) = lower('%q') "
+                                "LIMIT 1", // to avoid denial of service
+                                m_pszTableName);
+
+                            SQLResult oResult;
+                            err = SQLQuery(m_poDS->GetDB(), pszSQL, &oResult);
+                            sqlite3_free(pszSQL);
+                            if ( err == OGRERR_NONE && oResult.nRowCount == 1 )
+                            {
+                                const char *pszMetadata = SQLResultGetValue(&oResult, 0, 0);
+                                if( pszMetadata &&
+                                    STARTS_WITH_CI(pszMetadata, "coordinate_epoch=") )
```

Comment:
Is this required? It should be already covered by the SQL `md.metadata LIKE 'coordinate_epoch=%%'` clause already

Reply:
I'm kind of paranoid, but yes that isn't needed unless sqlite3 has bugs. Changing it into an assertion

Reply:
> unless sqlite3 has bugs

they don't have any, that's why they don't need a bug tracker :upside_down_face: